build.rs

```diff
@@ -0,0 +1,65 @@
+use std::env;
+use std::path::PathBuf;
+
+extern crate cmake;
+use cmake::Config;
+
+fn main() {
+    let dst = Config::new(".")
+        .define("BUILD_STATIC_LIBS", "ON")
+        .build_target("MQF")
+        .build();
+
+    // TODO: there are probably better ways to do this...
+    let target = env::var("TARGET").unwrap();
+    if target.contains("apple") {
+        println!("cargo:rustc-link-lib=dylib=c++");
+    } else if target.contains("linux") {
+        println!("cargo:rustc-link-lib=dylib=stdc++");
+    } else {
+        unimplemented!();
+    }
+
+    println!("cargo:rustc-link-search=native={}/build/src", dst.display());
+    println!("cargo:rustc-link-lib=static=MQF");
+
+    println!(
+        "cargo:rustc-link-search=native={}/build/ThirdParty/stxxl/lib",
+        dst.display()
+    );
+
+    let mode = match env::var("PROFILE").unwrap().as_ref() {
+        "debug" => "_debug",
+        _ => "",
+    };
+    println!("cargo:rustc-link-lib=static=stxxl{}", mode);
```

Comment:
@shokrof I can't skip linking stxxl here, since libMQF.a brings the on-disk MQF and if it is not included there are many symbols missing when I try to run the tests. Not sure how to (or even if it is desirable) to modularize what gets added to the static/dynamic library,

Reply:
Do you mean the c++ test?

Reply:
No, the Rust tests. The code builds, but when I try to link the static library to the Rust code (to run the tests) there are missing symbols.

Reply:
I can create a new target for you that only has MQF if it will make your life easier